finetune_adapter.py

```diff
@@ -64,7 +65,10 @@ def main(
 
     config = Config.from_name(name=checkpoint_dir.name, block_size=max_seq_length)
 
-    with EmptyInitOnDevice(device=fabric.device, dtype=torch.float32 if fabric._precision.precision == "32-true" else torch.bfloat16):
+    with EmptyInitOnDevice(
+        device=fabric.device,
+        dtype=torch.float32 if fabric.strategy.precision.precision == "32-true" else torch.bfloat16
```

Comment:
Fabric doesn't offer any mapping from precision str/class to torch.dtype. So any other values would need modification here

Reply:
One could also just use the local variable `precision` here.